oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneDocumentMaker.java

```diff
@@ -315,6 +312,29 @@ protected boolean indexTypeOrderedFields(Document doc, String pname, int tag, Pr
         }
         return fieldAdded;
     }
+    
+    private static BytesRef checkTruncateLength(String prop, String value, String path, int maxLength) {
+        String truncated = value;
+        if (value.length() > maxLength) {
+            log.warn("Truncating property {} having length {} at path:[{}] as it is > {}", prop, value.length(), path, maxLength);
+            truncated = value.substring(0, maxLength);
```

Comment:
This will not truncate correctly if there are surrogate pairs

Reply:
The truncation is on the original string so, it should be fine.

Reply:
Ah I understood now yes it's a problem for surrogate pairs